### PR TITLE
Extension try-catch

### PR DESCRIPTION
**Pull request by Delawen**
## Purpose

This is a step extension for the try-catch camel EIP.

**Is this a new step extension or an improvement for an existing one?**

New step extension

### What is the step(s) that this extension covers?

try-catch camel EIP

### Open Questions and Pre-Merge TODOs

Please ensure your pull request adheres to the following guidelines:

- [x] There are tests that cover at least 75% of the step extension code
- [x] All tests pass with `yarn test`
- [x] Improvements of existing step extensions should be backwards compatible unless specified otherwise.

### CODE REVIEW

File: try-catch-eip/src/DoCatch.tsx
Comment (Delawen) [inline try-catch-eip/src/DoCatch.tsx:2]: This is the last thing I need to fix. How¿?

Comment (Delawen) [reply try-catch-eip/src/DoCatch.tsx:2]: @kahboom

File: try-catch-eip/public/manifest.json
Comment (lordrip) [inline try-catch-eip/public/manifest.json:1]: Probably we should remove this file and both icons as well :)

**Review by lordrip — APPROVED** [review]
LGTM, Just a few pending strict comparison.

**Comment by lordrip** [comment]
Some thoughts:

Considering that we don't have the steps types at the moment, could be possible to type the steps array as an array? at least we would benefit from having `intellisense` and a few safe typings.

Also, maybe this is a little too far, but another extra safety measure that we could take is to type the arrays as `unknown[]`
to avoid accessing properties without verifying their existence first :thinking: 

LMYT :smiley:

File: try-catch-eip/src/App.css
Comment (Delawen) [reply try-catch-eip/src/App.css]: Removed

File: try-catch-eip/src/App.test.tsx
Comment (Delawen) [reply try-catch-eip/src/App.test.tsx:4]: I did a whole replacement of all `"` with `'`

File: try-catch-eip/public/manifest.json
Comment (Delawen) [reply try-catch-eip/public/manifest.json]: Done.

File: try-catch-eip/src/DoCatch.tsx
Comment (mmelko) [inline try-catch-eip/src/DoCatch.tsx:113]: This is not an API function, so it won't update kaoto. You should use `updateStep` instead. see https://github.com/KaotoIO/kaoto-ui/blob/main/src/components/VisualizationStepViews.tsx#L202

Comment (kahboom) [reply try-catch-eip/src/DoCatch.tsx:113]: We do have to move those functions outside to an easier place to reference them at some point, the problem is that for things like the `notifyKaoto` it has to be within a component, but I'm sure there is a workaround we can use.

**Comment by Delawen** [comment]
image

File: try-catch-eip/src/DoCatch.tsx
Comment (Delawen) [reply try-catch-eip/src/DoCatch.tsx]: Done.

**Comment by Delawen** [comment]
I think this is ready. In my local it works :smile:

**Comment by Delawen** [comment]
When merged, please merge https://github.com/KaotoIO/kaoto-viewdefinition-catalog/pull/16 too

**Review by mmelko — COMMENTED** [review]
would be possible to delete the kaoto/types folder ? I feel like we don;'t need to have whole kaoto here. I'd prefer rather to have the extension less clean with much less lines and files.

**Comment by Delawen** [comment]
I would delete them if we had a replacement in the form of some npm package.

**Comment by mmelko** [comment]
but no other extension has it currently. In theory it can bring some issues when kaoto code changes more drastically. Also I guess it's not synced with current Kaoto code base.

@kahboom wdyt  ? leave it there now ? or rather remove and not use typings as in other extensions ? Personally I'm for less code.

**Comment by kahboom** [comment]
I leave it to @Delawen whether to keep or remove it. If there will be continuous development on this extension then it could be useful to keep, but if not then deleting would be better. We don't have an npm package for this and probably won't, since we already meant to be using the MF types package but just haven't set it up yet (assuming it needs to be).

**Review by lordrip — APPROVED** [review]

**Pull request merged**
